### PR TITLE
Update GraalVM to 22.3.1

### LINKED ISSUE #5086: Update GraalVM to 22.3.1 to fix chrome inspector
This task is automatically imported from the old Task Issue Board and it was originally created by jaroslavtulach.
Original issue is [here](https://www.pivotaltracker.com/story/show/184322520).
<hr />

[GraalVM 22.3.1](https://www.pivotaltracker.com/n/projects/2539304/stories/182485940) has been officially released on Jan 25, 2023. 

It contains fix for [GR-5513](https://github.com/oracle/graal/issues/5513) which allows us to remove [hacks introduced](https://github.com/enso-org/enso/pull/3941/files#r1042169729) when working on #183855919

Another hack is in PR #3970

### PR DESCRIPTION
**Pull request by Akirathan**
### Pull Request Description
Updates the engine to GraalVM 22.3.1 version, which contains fixes for:
- Chrome inspector issues - https://github.com/enso-org/enso/pull/3941/commits/38eb3b5932d91fe858c9ba96b9a0543118bd2364

### Important Notes
- Update to GraalVM 22.3.1
- Remove host object wrapping workaround

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
> ...added clean build...

Yup, [that needs clean build](https://github.com/enso-org/enso/actions/runs/4135737874/jobs/7148609016#step:10:3701).

**Review by hubertp — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -560,6 +560,7 @@
 - [Profile engine startup][4110]
 - [Report type of polyglot values][4111]
 - [Engine can now recover from serialization failures][5591]
+- [Update to GraalVM 22.3.1][????]
```
Comment (radeusgd) [inline CHANGELOG.md]:
```suggestion
- [Update to GraalVM 22.3.1][5602]
```

**Review by radeusgd — APPROVED** [review]
So good to see this workaround gone! 🎉 

I assume you have tested that the new build works as expected in Chrome Inspector? Because I think we didn't have unit tests for it, so definitely would be good to verify if the fix was enough.

**Comment by Akirathan** [comment]
> I assume you have tested that the new build works as expected in Chrome Inspector? Because I think we didn't have unit tests for it, so definitely would be good to verify if the fix was enough.

I have verified that on debugging `Scatter_Plot_Spec.enso`. Also, we have some unit tests for that in https://github.com/enso-org/enso/blob/develop/engine/runtime/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java that even contained the check for the workaround that I had to remove in this PR.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">enso/DebuggingEnsoTest.java at develop · enso-org/enso</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

File: CHANGELOG.md (hunk 1 above)
Comment (Akirathan) [reply CHANGELOG.md]: Wow, CI job that checks that have not found that I had `???`. I guess that this was my bad. Next time I will remember to not touch the CHANGELOG until I have a PR. Thanks for noticing this.

Comment (radeusgd) [reply CHANGELOG.md]: I thought that the CI job was only checking if changelog was modified at all, I don't think it has any special logic.

**Comment by radeusgd** [comment]
> > I assume you have tested that the new build works as expected in Chrome Inspector? Because I think we didn't have unit tests for it, so definitely would be good to verify if the fix was enough.
> 
> I have verified that on debugging `Scatter_Plot_Spec.enso`. Also, we have some unit tests for that in [`develop`/engine/runtime/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java](https://github.com/enso-org/enso/blob/develop/engine/runtime/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java?rgh-link-date=2023-02-10T16%3A38%3A02Z) that even contained the check for the workaround that I had to remove in this PR.
> 
> >  GitHub**[enso/DebuggingEnsoTest.java at develop · enso-org/enso](https://github.com/enso-org/enso)**Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.

Ah sorry! I missed that. Looks amazing then 😁

File: engine/runtime/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java [hunk 2]
```diff
@@ -208,15 +205,12 @@ public void testHostValues() {
...
           DebugValue listValue = scope.getDeclaredValue("list");
-          // ArrayList is internally represented as Enso list, but as an object
-          // initialized in host context, it suffers from the issue mentioned in
-          // https://github.com/oracle/graal/issues/5513. Therefore, we display
-          // it just as 'HostObject' in the debugger.
           assertNotNull(listValue);
-          assertTrue(listValue.toDisplayString().startsWith("HostObject"));
+          assertTrue(listValue.isArray());
+          assertEquals(10, listValue.getArray().get(0).asInt());
+          assertEquals(20, listValue.getArray().get(1).asInt());
```
Comment (radeusgd) [inline engine/runtime/src/test/java/org/enso/interpreter/test/DebuggingEnsoTest.java:213]: 🎉 ❤️

**Review by JaroslavTulach — APPROVED** [review]
200 hacky lines deleted! Nice.

**Review by radeusgd — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]
:heart:

**Review by mwu-tow — APPROVED** [review]

**Pull request merged**